models/product/src/product-data/presets/boring-generic-milk-product-data.spec.ts

```diff
@@ -0,0 +1,67 @@
+import type { TProductDataGraphql, TProductDataRest } from '../types';
+import { graphqlPresets, restPresets } from '.';
+
+function validateRestModel(restModel: TProductDataRest) {
+  expect(restModel).toEqual(
+    expect.objectContaining({
+      name: expect.objectContaining({
+        en: 'Boring Generic Milk',
+        de: 'Langweilige generische Milch',
+      }),
+      description: expect.objectContaining({
+        en: 'Very average milk produced by a very average cow!',
+        de: 'Sehr durchschnittliche Milch von einer sehr durchschnittlichen Kuh!',
+      }),
+      slug: expect.objectContaining({
+        en: 'boring-generic-milk-slug',
+      }),
+    })
+  );
+}
+
+function validateGraphqlModel(graphqlModel: TProductDataGraphql) {
+  expect(graphqlModel).toEqual(
+    expect.objectContaining({
+      name: 'Boring Generic Milk',
+      nameAllLocales: expect.arrayContaining([
+        expect.objectContaining({
+          locale: 'en',
+          value: 'Boring Generic Milk',
+        }),
+      ]),
+      description: 'Very average milk produced by a very average cow!',
+      descriptionAllLocales: expect.arrayContaining([
+        expect.objectContaining({
+          locale: 'en',
+          value: 'Very average milk produced by a very average cow!',
+        }),
+        expect.objectContaining({
+          locale: 'de',
+          value:
+            'Sehr durchschnittliche Milch von einer sehr durchschnittlichen Kuh!',
+        }),
+      ]),
+      slug: 'boring-generic-milk-slug',
+      slugAllLocales: expect.arrayContaining([
+        expect.objectContaining({
+          locale: 'en',
+          value: 'boring-generic-milk-slug',
+        }),
+      ]),
+    })
+  );
+}
+
+describe('Product "boring generic milk" presets', () => {
```

Comment:
It would be great to also have tests for the compatibility presets.

Reply:
https://github.com/commercetools/test-data/pull/790/commits/36a06047a64fd86c14b636fd044b155a0205289f

Note: the test for graphql compat preset is marked as skipped. The main problem is that the the compat builder uses e.g. `name()` but the graphql builder expects `nameAllLocales()`. This leads to the name data to be randomly generated as the object passed to `name()` gets ignored. Therefore, the expected graphql object doesn't match.

We can discuss this case later if you want.